karapace/schema_models.py

```diff
@@ -51,85 +53,134 @@ def parse_jsonschema_definition(schema_definition: str) -> Draft7Validator:
     return Draft7Validator(schema)
 
 
-def parse_protobuf_schema_definition(schema_definition: str) -> ProtobufSchema:
+def parse_protobuf_schema_definition(
+    schema_definition: str,
+    references: Optional[List[Reference]] = None,
+    dependencies: Optional[Dict[str, Dependency]] = None,
+    validate_references: bool = True,
+) -> ProtobufSchema:
     """Parses and validates `schema_definition`.
 
     Raises:
-        Nothing yet.
+        ProtobufUnresolvedDependencyException if Protobuf dependency cannot be resolved.
 
     """
-
-    return ProtobufSchema(schema_definition)
-
-
-@unique
-class SchemaType(str, Enum):
-    AVRO = "AVRO"
-    JSONSCHEMA = "JSON"
-    PROTOBUF = "PROTOBUF"
+    protobuf_schema = ProtobufSchema(schema_definition, references, dependencies)
+    if validate_references:
+        result = protobuf_schema.verify_schema_dependencies()
+        if not result.result:
+            raise ProtobufUnresolvedDependencyException(f"{result.message}")
+    return protobuf_schema
 
 
 def _assert_never(no_return: NoReturn) -> NoReturn:
     raise AssertionError(f"Expected to be unreachable {no_return}")
 
 
+
 class TypedSchema:
-    def __init__(self, schema_type: SchemaType, schema_str: str):
+    def __init__(
+        self,
+        *,
+        schema_type: SchemaType,
+        schema_str: str,
+        schema: Optional[Union[Draft7Validator, AvroSchema, ProtobufSchema]] = None,
+        references: Optional[List[Reference]] = None,
+        dependencies: Optional[Dict[str, Dependency]] = None,
+    ):
         """Schema with type information
 
         Args:
             schema_type (SchemaType): The type of the schema
             schema_str (str): The original schema string
+            schema (Optional[Union[Draft7Validator, AvroSchema, ProtobufSchema]]): The parsed and validated schema
+             references (Optional[List[Dependency]]): The references of schema
         """
         self.schema_type = schema_type
-        self.schema_str = TypedSchema.normalize_schema_str(schema_str, schema_type)
+        self.references = references
+        self.dependencies = dependencies
+        self.schema_str = TypedSchema.normalize_schema_str(schema_str, schema_type, schema)
         self.max_id: Optional[SchemaId] = None
-
         self._fingerprint_cached: Optional[str] = None
+        self._str_cached: Optional[str] = None
 
     def to_dict(self) -> Dict[str, Any]:
         if self.schema_type is SchemaType.PROTOBUF:
             raise InvalidSchema("Protobuf do not support to_dict serialization")
-        return json_decode(self.schema_str, Dict[str, Any])
+        return json_decode(self.schema_str, Dict)
```

Comment:
```suggestion
        return json_decode(self.schema_str, Dict[str, Any])
```
I think this was a mistake from the merge from main.